here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/WriteFeatureApiTask.java

```diff
@@ -228,6 +238,202 @@ protected void init() {}
     }
   }
 
+  // TODO this might change if naksha_plpgsql.sql gets updated and the message is modified
+  private static final Pattern FEATURE_ID_FROM_ERR = Pattern.compile("The feature \\s*'([^']*)' uuid");
+  private static final int MAX_RETRY_ATTEMPT = 5;
+
+  private @NotNull XyzResponse executePatchFeatureById() throws JsonProcessingException {
+
+    final XyzFeature featureFromRequest = singleFeatureFromRequestBody();
+
+    final String spaceId = ApiParams.extractMandatoryPathParam(routingContext, SPACE_ID);
+    final String featureId = ApiParams.extractMandatoryPathParam(routingContext, FEATURE_ID);
+    final QueryParameterList queryParams = queryParamsFromRequest(routingContext);
+    final List<String> addTags = extractParamAsStringList(queryParams, ADD_TAGS);
+    final List<String> removeTags = extractParamAsStringList(queryParams, REMOVE_TAGS);
+
+    // Validate parameters
+    if (!featureId.equals(featureFromRequest.getId())) {
+      return verticle.sendErrorResponse(
+          routingContext,
+          XyzError.ILLEGAL_ARGUMENT,
+          "URI path parameter featureId is not the same as id in feature request body.");
+    }
+
+    // as applicable, modify features based on parameters supplied
+    addTagsToFeature(featureFromRequest, addTags);
+    removeTagsFromFeature(featureFromRequest, removeTags);
+
+    final List<XyzFeature> featuresFromRequest = new ArrayList<>();
+    featuresFromRequest.add(featureFromRequest);
+    List<String> featureIds = new ArrayList<>();
+    featureIds.add(featureId);
+    return attemptFeaturesPatching(spaceId, featureIds, featuresFromRequest, HttpResponseType.FEATURE, 0);
+  }
+
+  private XyzResponse attemptFeaturesPatching(
```

Comment:
This function uses `executeWriteRequestFromSpaceStorage()` and `executeReadRequestFromSpaceStorage()` inherited from `AbstractApiTask` class. I'd prefer not to move this to a helper class to avoid some unnecessary refactoring. Or is there an easy way that I don't know of?